src/rt/ds/hashmap.h

```diff
@@ -351,10 +411,15 @@ namespace verona::rt
       return false;
     }
 
-    void insert_unique(Alloc* alloc, Entry& entry)
+    bool insert(Alloc* alloc, Entry entry)
     {
       size_t dummy;
-      auto unique = insert(alloc, entry, dummy);
+      return insert(alloc, std::move(entry), dummy);
+    }
+
+    void insert_unique(Alloc* alloc, Entry entry)
+    {
+      auto unique = insert(alloc, std::move(entry));
```

Comment:
This feels a bit suspicious.   The Entry has been copied into this call, and then moved?  Given you are using callbacks rather than destructors now, is this even needed now?

Reply:
Also, this is going to call `on_insert` internally, which feels like it is going to have called that repeatedly?

Reply:
`on_insert` is only called when the entry is inserted, not when updated. I'll add a note about that in the docs